packages/iris-grid/src/IrisGridTableModel.js

```diff
@@ -458,8 +458,36 @@ class IrisGridTableModel extends IrisGridModel {
     return '';
   }
 
+  getColumnsWithLayoutHints = memoize((columns, hints) => {
+    if (hints) {
+      const frontColumns = new Array(hints.frontColumns.length);
+      const backColumns = new Array(hints.backColumns.length);
+      const middleColumns = [];
+
+      columns.forEach(col => {
```

Comment:
You could just allocate the array once and fill it in. eg.

```
const newColumns = new Array(columns.length);
let middleIndex = hints.frontColumns.length;
forEach...
if (frontColumn) {
  newColumns[indexOf] = col
} else if (backColumn) {
  // Get the correct back index
  newColumns[columns.length - indexOf] = col
} else {
  newColumns[middleIndex] = col;
  middleIndex += 1;
}
```
Also instead of doing `includes` and `indexOf`, which would require iterating the array twice (unless there's compiler optimization magic?), could just get the indexOf right away, and check if indexOf >= 0.